azurerm/resource_arm_backup_protected_file_share.go

```diff
@@ -0,0 +1,292 @@
+}
+
+func resourceArmBackupProtectedFileShareCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).RecoveryServices.ProtectedItemsClient
+	opClient := meta.(*ArmClient).RecoveryServices.BackupOperationStatusesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	vaultName := d.Get("recovery_vault_name").(string)
+	storageAccountID := d.Get("source_storage_account_id").(string)
+	fileShareName := d.Get("source_file_share_name").(string)
+	policyID := d.Get("backup_policy_id").(string)
+
+	//get storage account name from id
+	parsedStorageAccountID, err := azure.ParseAzureResourceID(storageAccountID)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Unable to parse source_storage_account_id '%s': %+v", storageAccountID, err)
+	}
+	accountName, hasName := parsedStorageAccountID.Path["storageAccounts"]
+	if !hasName {
+		return fmt.Errorf("[ERROR] parsed source_storage_account_id '%s' doesn't contain 'storageAccounts'", storageAccountID)
+	}
+
+	protectedItemName := fmt.Sprintf("AzureFileShare;%s", fileShareName)
+	containerName := fmt.Sprintf("StorageContainer;storage;%s;%s", parsedStorageAccountID.ResourceGroup, accountName)
+
+	log.Printf("[DEBUG] Creating/updating Recovery Service Protected File Share %q (Container Name %q)", protectedItemName, containerName)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err2 := client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+		if err2 != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Recovery Service Protected File Share %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err2)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_backup_protected_file_share", *existing.ID)
+		}
+	}
+
+	item := backup.ProtectedItemResource{
+		Properties: &backup.AzureFileshareProtectedItem{
+			PolicyID:          &policyID,
+			ProtectedItemType: backup.ProtectedItemTypeAzureFileShareProtectedItem,
+			WorkloadType:      backup.DataSourceTypeAzureFileShare,
+			SourceResourceID:  utils.String(storageAccountID),
+			FriendlyName:      utils.String(fileShareName),
+		},
+	}
+
+	resp, err := client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, item)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Recovery Service Protected File Share %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+	}
+
+	locationURL, err := resp.Response.Location()
+	if err != nil || locationURL == nil {
+		return fmt.Errorf("Error creating/updating Azure File Share backup item %q (Vault %q): Location header missing or empty", containerName, vaultName)
+	}
+
+	opResourceID := azure.HandleAzureSdkForGoBug2824(locationURL.Path)
```

Comment:
I believe the `HandleAzureSdkForGoBug2824` isn't required anymore (the service team claims its been fixed https://github.com/Azure/azure-sdk-for-go/issues/2824#issuecomment-563188743)